app/routes.py

```diff
@@ -13,18 +14,23 @@
 
 # Creates a dictionary from the adopter request
 def get_dict_from_request(required_fields, optional_fields):
+    #We do this rather than use request.json because OC prior to 12.12/13.7/14.0
+    # *claimed* to be sending UTF-8, but was actually sending latin-1 (or system default)
+    # This doesn't seem to break anything when tested against the fixed version, but more non-(US/EU)
+    # locales may still be broken if the system default encoding is sufficiently different
+    req_json_bytes = request.data
+    req_json = json.loads(req_json_bytes.decode("latin-1").encode("utf-8"))
```

Comment:
Couldn't this have side effects for some characters where `latin1` and `utf8` overlap?
According to your comment, you are unsure yourself :D
Something like this would be safer:
```py
try:
  req_json = json.loads(req_json_bytes)
catch UnicodeDecodeError:
  req_json = json.loads(req_json_bytes.decode("latin-1"))
```

Reply:
Good point, added.